src/pkcs11/pkcs11-spy.c

```diff
@@ -969,6 +969,15 @@ C_SignInit(CK_SESSION_HANDLE hSession, CK_MECHANISM_PTR pMechanism, CK_OBJECT_HA
 	enter("C_SignInit");
 	spy_dump_ulong_in("hSession", hSession);
 	fprintf(spy_output, "pMechanism->type=%s\n", lookup_enum(MEC_T, pMechanism->mechanism));
+	if (pMechanism->pParameter != NULL) { /* XXX assuming PSS parameter */
+		CK_RSA_PKCS_PSS_PARAMS *param =
+			(CK_RSA_PKCS_PSS_PARAMS *) pMechanism->pParameter;
```

Comment:
Yes, we should. But since this was the first mechanism using parameter, I wanted to avoid listing all the PSS mechanisms. But as we will have OEAP too, we should certainly do that. I will submit an update tomorrow. It is quite late here in Europe.

Reply:
thanks

Reply:
What other parameter types are possible in `C_SignInit`? Certainly not OAEP (which belongs to `C_DecryptInit`).

Reply:
oh ... my bad. It was very late yesterday. But reading through the PKCS#11 mechanisms specification, we can see that for example [2.8.9 General-length AES-MAC](http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cs01/pkcs11-curr-v2.40-cs01.html#_Toc399398925) is a mechanism that has parameters and is using/can use the `C_Sign` interface.

Reply:
OK, thanks. Understood. Made the same change for OAEP in ...spy.

Reply:
FYI done. All checks passed.